terraform/ec2/amis.tf

```diff
@@ -89,46 +89,137 @@ EOF
+      instance_type      = "t4g.nano"
+      user_data          = <<EOF
+#! /bin/bash
+sudo snap refresh amazon-ssm-agent
+EOF
+    }
+    ubuntu22 = {
+      os_family          = "ubuntu"
+      ami_search_pattern = "ubuntu/images/hvm-ssd/ubuntu-jammy*"
+      ami_owner          = "aws-marketplace"
+      ami_product_code   = []
+      family             = "debian"
+      arch               = "x86_64"
+      login_user         = "ubuntu"
+      user_data          = <<EOF
+#! /bin/bash
+sudo snap refresh amazon-ssm-agent
+EOF
+    }
+    arm_ubuntu22 = {
+      os_family          = "ubuntu"
+      ami_search_pattern = "ubuntu/images/hvm-ssd/ubuntu-jammy*"
+      ami_owner          = "aws-marketplace"
+      ami_product_code   = []
+      family             = "debian"
+      arch               = "arm64"
+      login_user         = "ubuntu"
+      instance_type      = "t4g.nano"
+      user_data          = <<EOF
+#! /bin/bash
+sudo snap refresh amazon-ssm-agent
+EOF
+    }
+    # Debian Distribution
+    debian11 = {
+      os_family          = "debian"
+      ami_search_pattern = "debian-11-*"
+      ami_owner          = "aws-marketplace"
+      ami_product_code   = []
+      family             = "debian"
+      arch               = "x86_64"
+      login_user         = "admin"
+      user_data          = <<EOF
+#! /bin/bash
+cd /tmp
+sudo wget https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/debian_amd64/amazon-ssm-agent.deb
+while sudo fuser {/var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock,/var/lib/dpkg/lock-frontend}; do
+   echo 'Waiting for dpkg lock...' && sleep 1
+done
+sudo dpkg -i amazon-ssm-agent.deb
+sudo systemctl enable amazon-ssm-agent
+EOF
+    }
+    arm_debian11 = {
+      os_family          = "debian"
+      ami_search_pattern = "debian-11-*"
+      ami_owner          = "aws-marketplace"
+      ami_product_code   = []
+      family             = "debian"
+      arch               = "arm64"
+      login_user         = "admin"
+      instance_type      = "t4g.nano"
+      user_data          = <<EOF
+#! /bin/bash
+cd /tmp
+sudo wget https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/debian_amd64/amazon-ssm-agent.deb
+while sudo fuser {/var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock,/var/lib/dpkg/lock-frontend}; do
+   echo 'Waiting for dpkg lock...' && sleep 1
+done
+sudo dpkg -i amazon-ssm-agent.deb
+sudo systemctl enable amazon-ssm-agent
 EOF
     }
     debian10 = {
       os_family          = "debian"
-      ami_search_pattern = "debian-10-amd64*"
-      ami_owner          = "679593333241"
-      ami_id             = "ami-0900b247bf638c13f"
-      # NOTE: we need product code to pick the right debian 10.
-      ami_product_code = [
-      "auhljmclkudu651zy27rih2x2"]
-      family     = "debian"
-      arch       = "amd64"
-      login_user = "admin"
-      user_data  = <<EOF
+      ami_search_pattern = "debian-10-*"
+      ami_owner          = "aws-marketplace"
+      ami_product_code   = []
```

Comment:
The previous version had a comment about the product code. In your testing did the "correct" debian10 get selected?

Reply:
As we use `aws-marketplace` as the `ami_owner`, they have the supported debian images. SO product code is not required now

Reply:
<img width="739" alt="Screen Shot 2022-10-04 at 11 33 33 AM" src="https://user-images.githubusercontent.com/28486681/193898078-f75a9f8c-64fe-4fd6-aa02-c1716ecaed9c.png">

<img width="769" alt="Screen Shot 2022-10-04 at 11 35 58 AM" src="https://user-images.githubusercontent.com/28486681/193898489-428960b3-3673-4f34-8498-0c2d946e91fe.png">